### ISSUE TITLE
Off-by-one error in final output symlink for dbm

### ISSUE THREAD

**Issue by araikes** [CoBrALab/optimized_antsMultivariateTemplateConstruction#32]
DBM fails out of the gate. Script expects final non-linear output to be in "output/final" where as the file tree is e.g., "output/nlin/3"

https://github.com/CoBrALab/optimized_antsMultivariateTemplateConstruction/blob/d3b34c3564ca7a74c59bb22070f2ae92b236a529/dbm.sh#L380-L381

**Comment by gdevenyi** [comment]
Note there's a small change to the modelbuild.sh file:
https://github.com/CoBrALab/optimized_antsMultivariateTemplateConstruction/pull/31/files#diff-52fbb57ee96676caf18945989c666ec63f63fc70d470d652d406efc37fe836abR1047

**Comment by araikes** [comment]
Small point but may be important depending on who is using it .... that change to modelbuild.sh works as long as you're not doing so inside a container, otherwise it is just an empty link.

**Comment by gdevenyi** [comment]
Really? That doesn't make sense to me as it is a relative symlink. Can you test a manual running of
```
ln -srf output/nlin/3 output/final
```

within a container?

**Comment by araikes** [comment]
I'll check it when I get back to my computer. It occurs to me though that I mounted the template folder for more generalizable code across studies, and so that pathing doesn't exist for the symbolic link if I then run DBM in a separate instance.

**Comment by gdevenyi** [comment]
This is what the current config is based on:
```
'-r'
‘--relative’
Make symbolic links relative to the link location. This option is only valid with the --symbolic option.

‘-s’
‘--symbolic’
Make symbolic links instead of hard links. This option merely produces an error message on systems that do not support symbolic links.
```

So as I understand (and experience on my systems) that `ln` traverses down into `output` and generates a relative symlink between `nlin/3` and `final`

**Comment by araikes** [comment]
I think the problem is here:

https://github.com/CoBrALab/optimized_antsMultivariateTemplateConstruction/blob/d3b34c3564ca7a74c59bb22070f2ae92b236a529/modelbuild.sh#L1039-L1041

Pretending that i = 4, at the end of the while loop you increment `i` by 1 so that `i = 4`. Therefore the symlinking attempts to link `output/nlin/4` to `output/final` when `output/nlin/3` is the final.

I've tested the symlinking in a shell using the same volume bindings and it works fine when I point it to `output/nlin/3`

**Comment by gdevenyi** [comment]
Ahh, off by one error :+1:

Will sort that.

My testing of the dbm was all post-hoc on an already done modelbuild where I manually (correctly generated a link)

**Comment by gdevenyi** [comment]
Fixed in #31
